src/pybind/mgr/nfs/cluster.py

```diff
@@ -55,27 +56,32 @@ def __init__(self, mgr: 'Module') -> None:
     def _call_orch_apply_nfs(
             self,
             cluster_id: str,
-            placement: Optional[str],
+            placement: Optional[str] = None,
             virtual_ip: Optional[str] = None,
+            ingress_mode: Optional[IngressType] = None,
             port: Optional[int] = None,
     ) -> None:
         if not port:
             port = 2049   # default nfs port
         if virtual_ip:
             # nfs + ingress
             # run NFS on non-standard port
+            assert ingress_mode is not None
             spec = NFSServiceSpec(service_type='nfs', service_id=cluster_id,
                                   placement=PlacementSpec.from_string(placement),
                                   # use non-default port so we don't conflict with ingress
-                                  port=10000 + port)   # semi-arbitrary, fix me someday
+                                  port=10000 + port,  # semi-arbitrary, fix me someday
+                                  virtual_ip=virtual_ip.split('/')[0] if ingress_mode == IngressType.keepalive_only else None)
             completion = self.mgr.apply_nfs(spec)
             orchestrator.raise_if_exception(completion)
             ispec = IngressSpec(service_type='ingress',
                                 service_id='nfs.' + cluster_id,
                                 backend_service='nfs.' + cluster_id,
+                                placement=PlacementSpec.from_string(placement),
                                 frontend_port=port,
```

Comment:
On the Ingress frontend port: 
If we're deploying `keepalived only`, the frontend port is not used (no haproxy here), hence having the same port for both the ingress daemon and the nfs backend results in an issue because there's something trying to allocate the same port and preventing one of the two daemons being deployed.
```
If keepalived_only: 
 frontend_port +=(rand() % 10)
```

or
... just ignore that param since it's not strictly required.

Reply:
I have it so it now sets to `None` for keepalive only ingress (which should be fine as keepalive doesn't use it)